src/toil/jobStores/fileJobStore.py

```diff
@@ -280,27 +285,47 @@ def _supportsUrl(cls, url, export=False):
         return url.scheme.lower() == 'file'
 
     def writeFile(self, localFilePath, jobStoreID=None):
-
-        # log the name of the function writing the file in (the job creating it)
+        # Record the name of the job/function writing the file in the file name
         try:
-            sourceFunctionName = traceback.extract_stack()[0][3].split("(")[0]
+            # It ought to be third-to-last on the stack, above us and the FileStore. Probably.
+            sourceFunctionName = traceback.extract_stack()[-3][2]
         except:
-            sourceFunctionName = "x"
+            sourceFunctionName = "UNKNOWNJOB"
             # make sure the function name fetched has no spaces or oddities
+        finally:
+            pass
         if re.match("^[A-Za-z0-9_-]*$", sourceFunctionName):
             pass
         else:
-            sourceFunctionName = "x"
+            sourceFunctionName = "ODDLYNAMEDJOB"
         absPath = self._getUniqueName(localFilePath, jobStoreID, sourceFunctionName)
+        relPath = self._getRelativePath(absPath)
         shutil.copyfile(localFilePath, absPath)
-        return self._getRelativePath(absPath)
+        return relPath
 
     @contextmanager
     def writeFileStream(self, jobStoreID=None):
-        fd, absPath = self._getTempFile(jobStoreID)
+        # Record the name of the job/function writing the file in the file name
+        try:
```

Comment:
This looks like duplicated code from the above and should probably be its own function.